packages/Uploader/src/components/File/File.js

```diff
@@ -71,52 +70,83 @@ function File(props) {
+                >
+                  <TimesIcon color={tokens.textColor.icon} />
+                </Button.Icon>
+              )}
             </Popover.Trigger>
             <Popover.Content>
-              <Popover.Card>{I18n.t("uploader.cancel_upload")}</Popover.Card>
+              <Popover.Card>{I18n.t("uploader.cancel_upload", { name })}</Popover.Card>
             </Popover.Content>
           </Popover>
         );
     }
   }
 
-  function getProgressText() {
+  function getProgressText(showA11yProgress) {
+    const errorMessage = typeof onError === "function" ? onError(error) : error;
     switch (status) {
       case types.status.ERROR:
-        return typeof onError === "function" ? onError(error) : error;
+        return showA11yProgress ? I18n.t("uploader.progress.error", { name, error: errorMessage }) : errorMessage;
       case types.status.SUCCESS:
-        return I18n.t("uploader.progress.complete");
+        return showA11yProgress
+          ? I18n.t("uploader.progress.file_progress", { name, progress: I18n.t("uploader.progress.complete") })
+          : I18n.t("uploader.progress.complete");
       case types.status.CANCEL:
-        return I18n.t("uploader.progress.cancelled");
+        return showA11yProgress
+          ? I18n.t("uploader.progress.file_progress", { name, progress: I18n.t("uploader.progress.cancelled") })
+          : I18n.t("uploader.progress.cancelled");
       case types.status.IDLE:
-        return I18n.t("uploader.progress.idle");
+        return showA11yProgress
+          ? I18n.t("uploader.progress.file_progress", { name, progress: I18n.t("uploader.progress.idle") })
+          : I18n.t("uploader.progress.idle");
       default:
-        return I18n.t("uploader.progress.uploading", { progressWithUnits, sizeWithUnits });
+        return showA11yProgress
+          ? I18n.t("uploader.progress.file_progress", {
+              name,
+              progress: I18n.t("uploader.progress.uploading_percent", { a11yProgress }),
+            })
+          : I18n.t("uploader.progress.uploading", { progressWithUnits, sizeWithUnits });
     }
   }
 
   return (
-    <sc.File>
-      <sc.Left>
-        <sc.Info>
-          <sc.Name>{name}</sc.Name>
-          <sc.ProgressText status={status}>{getProgressText()}</sc.ProgressText>
-        </sc.Info>
-        <sc.ProgressBarWrapper>
-          <sc.ProgressBar data-pka-anchor="uploader-file-progressBar" progress={progress} status={status} />
-        </sc.ProgressBarWrapper>
-      </sc.Left>
-      <sc.Right status={status}>{renderIcon()}</sc.Right>
-    </sc.File>
+    <sc.FileListItem>
+      <div role="status" aria-atomic="true" aria-label={getProgressText(true)} />
+      <sc.File
+        aria-label={name}
+        aria-valuemin={0}
+        aria-valuemax={100}
+        aria-valuenow={progress}
+        aria-valuetext={status === types.status.ERROR ? I18n.t("uploader.status.error") : ""}
+        role="progressbar"
+        tabIndex={0}
```

Comment:
This is getting really close! But this `tabIndex` is still bugging me, we _shouldn't_ need it but it is odd that it doesn't behave the same if you remove it!

I farted around with it for a while, and I think I've got it now.  Take a look at this, and see if it works for you too:
https://github.com/acl-services/paprika/compare/UXD-1214-uploader-component-a11y-issues..UXD-1214-uploader-component-a11y-issues--aria-live-experiment